torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp

```diff
@@ -187,46 +192,76 @@ void initNvFuserPythonBindings(PyObject* module) {
           py::arg("ndims"),
           py::arg("dtype") = torch::jit::fuser::cuda::DataType::Float,
           py::return_value_policy::reference)
-      .def_static(
+      .def(
           "define_tensor",
-          [](size_t ndims,
-             std::vector<bool> contiguity,
+          [](FusionDefinitionContextManager& self,
+             std::vector<int> sizes,
+             std::vector<int> strides,
              torch::jit::fuser::cuda::DataType dtype =
                  torch::jit::fuser::cuda::DataType::Float) -> TensorView* {
             TORCH_CHECK(
-                ndims == contiguity.size(),
-                "The number of dimensions specified does not match contiguity.");
-            return TensorViewBuilder()
-                .ndims(ndims)
-                .dtype(dtype)
-                .contiguity(std::move(contiguity))
-                .build();
+                sizes.size() == strides.size(),
+                "The number of sizes does not match the number of strides.",
+                sizes.size(),
+                strides.size());
+
+            std::vector<IterDomain*> domain_sizes;
+            for (const auto i : c10::irange(sizes.size())) {
+              if (sizes[i] == 1) {
+                domain_sizes.push_back(IrBuilder::create<IterDomain>(
+                    self.fusionPtr()->zeroVal(),
+                    self.fusionPtr()->oneVal(),
+                    ParallelType::Serial,
+                    IterType::BroadcastWithStride));
+              } else {
+                domain_sizes.push_back(IrBuilder::create<IterDomain>(
+                    self.fusionPtr()->zeroVal(), IrBuilder::create<Int>()));
+              }
+            }
+
+            std::vector<bool> contig_info(strides.size(), false);
+            for (int i = contig_info.size() - 1; i >= 0; --i) {
+              if (i == static_cast<int>(contig_info.size() - 1)) {
+                contig_info[i] = (strides[i] == 1);
+              } else {
+                contig_info[i] =
+                    (strides[i] == (strides[i + 1] * sizes[i + 1]));
+              }
+            }
+
+            return IrBuilder::create<TensorView>(
+                IrBuilder::create<TensorDomain>(domain_sizes, contig_info),
+                dtype);
           },
-          py::arg("ndims"),
-          py::arg("contiguity"),
+          py::arg("sizes"),
+          py::arg("strides"),
           py::arg("dtype") = torch::jit::fuser::cuda::DataType::Float,
           py::return_value_policy::reference)
-      .def_static(
+      .def(
           "define_constant",
-          [](double val) -> torch::jit::fuser::cuda::Val* {
+          [](FusionDefinitionContextManager& self,
```

Comment:
Does this change imply that `define_constant` & `define_scalar` weren't functional? I don't see we added tests for those in this PR neither :stuck_out_tongue_winking_eye:

Reply:
`define_constant` and `define_scalar` were perviously added and they are functional in the sense that they can do something.  Tests just haven't been added in the `prims` world.